libvast/vast/logger.hpp

```diff
@@ -15,189 +15,69 @@
 
 #include "vast/config.hpp"
 #include "vast/detail/discard.hpp"
-#include "vast/detail/pp.hpp"
-#include "vast/detail/type_traits.hpp"
-
-#include <caf/detail/pretty_type_name.hpp>
-#include <caf/logger.hpp>
+#include "vast/error.hpp"
+
+// VAST_INFO -> spdlog::info
+// VAST_VERBOSE -> spdlog::debug
+// VAST_DEBUG -> spdlog::trace
+// VAST_TRACE -> spdlog::trace
+
+#if VAST_LOG_LEVEL == VAST_LOG_LEVEL_TRACE
+#  define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_TRACE
+#elif VAST_LOG_LEVEL == VAST_LOG_LEVEL_DEBUG
+#  define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_TRACE
+#elif VAST_LOG_LEVEL == VAST_LOG_LEVEL_VERBOSE
+#  define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_DEBUG
+#elif VAST_LOG_LEVEL == VAST_LOG_LEVEL_INFO
+#  define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_INFO
+#elif VAST_LOG_LEVEL == VAST_LOG_LEVEL_WARNING
+#  define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_WARN
+#elif VAST_LOG_LEVEL == VAST_LOG_LEVEL_ERROR
+#  define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_ERROR
+#elif VAST_LOG_LEVEL == VAST_LOG_LEVEL_CRITICAL
+#  define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_CRITICAL
+#elif VAST_LOG_LEVEL == VAST_LOG_LEVEL_QUIET
+#  define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_OFF
+#endif
+
+// Important: keep that below the log level mapping
+#include "vast/detail/logger.hpp"
+#include "vast/detail/logger_formatters.hpp"
+
+#define VAST_TRACE(...) SPDLOG_LOGGER_TRACE(vast::detail::logger(), __VA_ARGS__)
```

Comment:
We lost the ability to use trace for marking ENTER and EXIT of scopes. Can you add this back?